src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -2,18 +2,131 @@
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
 
+    public static final int INITIAL_CAPACITY = 16;
+    public static final double LOAD_FACTOR = 0.75;
+    public static final int GROWTH_FACTOR = 2;
+
+    private int size;
+    private Node<K, V>[] storage;
+
     @Override
     public void put(K key, V value) {
+        if (size == 0) {
+            storage = new Node[INITIAL_CAPACITY];
+        }
+
+        int hash = keyHash(key);
+        int storageIndex = storageIndex(hash, storage.length);
 
+        Node<K, V> currentNode = storage[storageIndex];
+        Node<K, V> prevNode = null;
+
+        while (true) {
```

Comment:
The use of while (true) can make the code less readable and potentially error-prone, as it relies on breaks or returns within the loop for termination. It's often better to use a more explicit condition for the loop to improve readability and reduce the risk of infinite loops. For instance, in the context of iterating over linked nodes, you could use while (currentNode != null) as the loop condition, which directly reflects the intention of traversing the list.